infra/Pos/Util/LogSafe.hs

```diff
@@ -52,13 +111,207 @@ logNoticeS  = logMessageS Notice
 logWarningS = logMessageS Warning
 logErrorS   = logMessageS Error
 
--- | Same as 'logMesssage', but log to two loggers, put only insecure
--- version to memmode.
+-- | Same as 'logMesssage', but log to secret logs, put only insecure
+-- version to memmode (to terminal).
 logMessageS
     :: (HasLoggerName m, MonadIO m)
     => Severity
     -> Text
     -> m ()
-logMessageS severity t = execSecureLogWrapped $ do
-    name <- getLoggerName
-    dispatchMessage name severity t
+logMessageS severity t =
+    reify selectSecretLogs $ \s ->
+    execSecureLogWrapped s $ do
+        name <- getLoggerName
+        dispatchMessage name severity t
+
+----------------------------------------------------------------------------
+-- Secure buildables
+----------------------------------------------------------------------------
+
+-- * Secutity log levels
+
+-- Stuff below provides way to use different 'instance Buildable' for writing
```

Comment:
Minor: not valid Haddock, you probably wanted to write `@instance 'Buildable'@` instead.